test/checkChallenge.test.js

```diff
@@ -0,0 +1,1627 @@
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 1, period_time: 80 } });
+
+      config.minperiodtimelive = 15;  // 15 seconds per stone * 5 stones = 1 minutes 15 seconds for all the 5 stones
+      config.maxperiodtimelive = 300; // 5  minutes per stone * 5 stones = 25 minutes           for all the 5 stones
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+
+    it('accept period time for all the stones live between min and max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 1, period_time: 80 } });
+
+      config.minperiodtimelive = 15;  // 15 seconds per stone * 5 stones = 1 minutes 15 seconds for all the 5 stones
+      config.maxperiodtimelive = 300; // 5  minutes per stone * 5 stones = 25 minutes           for all the 5 stones
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+
+    it('accept period time for all the stones live edge max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 1, period_time: 80 } });
+
+      config.minperiodtimelive = 15;  // 15 seconds per stone * 5 stones = 1 minutes 15 seconds for all the 5 stones
+      config.maxperiodtimelive = 300; // 5  minutes per stone * 5 stones = 25 minutes           for all the 5 stones
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+
+    it('reject period time for all the stones live too high', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 1, period_time: 1800 } });
+
+      config.minperiodtimelive = 10;
+      config.maxperiodtimelive = 300;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum Period Time for all the 5 stones for live games in canadian is 25 minutes, please reduce Period Time for all the 5 stones, or change the number of stones per period.' }));
+    });
+
+  });
+
+  describe('Fischer time settings', () => {
+
+    // sample:
+    // {"system":"fischer","time_control":"fischer","speed":"live","pause_on_weekends":false,"time_increment":10,"initial_time":80,"max_time":120}
+
+    // Just making sure it works similarly as byoyomi, so testing only "live" speed.
+
+    // Main Time 1 (Min Time)
+
+    it('reject main time 1 (min time) too low', () => {
```

Comment:
but it's not how we test it
currently we test:

- min time -> do not reject if mintime is between minmaintime and maxmaintime
- if min time is OK -> maxtime -> do not reject if maxtime is between minmaintime and maxmaintime
- etc. for increment time

now that you mention it i actually dont like it, because for example 30 minutes of byoyomi main time has nothing in common with 10 minutes of fischer max time, the 30 minutes of byoyomi will eventually dry out with mini periodtime, but the 30 minutes of fischer can continue indefinitely